src/appleseed/foundation/image/icanvas.h

```diff
@@ -198,10 +198,12 @@ template <typename T>
 inline void ICanvas::set_pixel(
     const size_t        x,
     const size_t        y,
-    const T             components[])
+    T                   components[])
 {
     const CanvasProperties& props = properties();
 
+    assert(sizeof(T) == props.m_pixel_size);
```

Comment:
This assertion is incorrect in multiple ways:

1. T is the type for one channel, not the whole pixel, hence `sizeof(T)` is typically not equal to `m_pixel_size` (maybe you meant `m_channel_size`?).
2. `set_pixel()` does pixel format conversion using `Pixel::convert_to_format()` so we must not assume that pixel formats match.